File: app/src/main/java/com/worldventures/dreamtrips/wallet/domain/entity/SmartCardStatus.java
package com.worldventures.dreamtrips.wallet.domain.entity;

import org.immutables.value.Value;

@Value.Immutable
public abstract class SmartCardStatus {
//
//   @Value.Default
//   public String sdkVersion() {
//      return "";
//   }
//
//   @Nullable
//   public abstract SmartCardFirmware firmwareVersion();

   @Value.Default
   public ConnectionStatus connectionStatus() {
      return ConnectionStatus.DISCONNECTED;
   }

   @Value.Default
   public boolean stealthMode() {
      return false;
   }

   @Value.Default
   public boolean lock() {
      return false;
   }

   @Value.Default
   public int batteryLevel() {
      return 0;
   }
//
//   @Value.Default
//   public long disableCardDelay() {
//      return 0; //never
//   }
//
//   @Value.Default
//   public long clearFlyeDelay() {
//      return 0; //never
//   }
}

File: wallet/src/main/java/com/worldventures/wallet/domain/entity/SmartCardStatus.kt
package com.worldventures.wallet.domain.entity

/**
 * @property firmwareVersion is version of nordic firmware. It can be empty on old firmware versions.
 * Property will be available when connectionStatus is CONNECTED.
 */
data class SmartCardStatus(
      val connectionStatus: ConnectionStatus = ConnectionStatus.DISCONNECTED,
      val stealthMode: Boolean = false,
      val lock: Boolean = false,
      val batteryLevel: Int = 0,
      val firmwareVersion: String = "",
      val disableCardDelay: Long = 0,
      val clearFlyeDelay: Long = 0
)
